nuxeo-drive-client/nxdrive/client/remote_document_client.py

```diff
@@ -205,22 +206,26 @@ def stream_update(
             op_name, file_path, filename=filename, mime_type=mime_type, **params)
 
     def delete(self, ref, use_trash=True):
-        op_input = "doc:" + self._check_ref(ref)
+        op_input = 'doc:' + self._check_ref(ref)
         if use_trash:
             try:
-                return self.execute("Document.SetLifeCycle", op_input=op_input,
-                                     value='delete')
+                if not version_lt('9.10', Options.server_version):
+                    return self.execute('Document.SetLifeCycle',
+                                        op_input=op_input, value='delete')
+                else:
+                    return self.execute('Document.Trash', op_input=op_input)
             except urllib2.HTTPError as e:
-                if e.code == 500:
-                    return self.execute("Document.Delete", op_input=op_input)
-                raise
+                if e.code != 500:
+                    raise
+        return self.execute('Document.Delete', op_input=op_input)
+
+    def undelete(self, uid):
+        op_input = 'doc:' + uid
+        if not version_lt('9.10', Options.server_version):
```

Comment:
In fact https://jira.nuxeo.com/browse/NXP-24282 has not been backported to 9.10 yet and I don't really know if it will be.